enterprise/views.py

```diff
@@ -983,22 +985,22 @@ def get_enterprise_program_enrollment_page(self, request, enterprise_customer, p
             purchase_action = self.actions['purchase_program']
             item = self.items['program_enrollment']
 
-        # Add any warning messages.
+        # Add any DSC warning messages.
         program_data_sharing_consent = get_data_sharing_consent(
             request.user.username,
             enterprise_customer.uuid,
             program_uuid=program_details['uuid'],
         )
         if program_data_sharing_consent.exists and not program_data_sharing_consent.granted:
-            add_consent_declined_message(request, enterprise_customer, program_title)
+            messages.add_consent_declined_message(request, enterprise_customer, program_title)
 
         discount_data = program_details.get('discount_data', {})
-        if discount_data.get('total_incl_tax_excl_discounts') is None:
-            add_missing_price_information_message(request, program_title)
-
         one_click_purchase_eligibility = program_details.get('is_learner_eligible_for_one_click_purchase', False)
+        # The following messages shouldn't both appear at the same time, and we prefer the eligibility message.
         if not one_click_purchase_eligibility:
-            add_not_one_click_purchasable_message(request, enterprise_customer, program_title)
+            messages.add_unenrollable_item_message(request, 'program')
+        elif discount_data.get('total_incl_tax_excl_discounts') is None:
+            messages.add_missing_price_information_message(request, program_title)
```

Comment:
Ian noted in UI/UX review that we should not have both of these messages be displayed at the same time, as that may confuse the user.